DisjointSet.java

```diff
@@ -0,0 +1,47 @@
+
+public class DisjointSet {
+	int[] parent;
+	int[] rank;
+
+	public DisjointSet(int numElements) {
+		parent = new int[numElements];
+		for (int i = 0; i < numElements; i++) {
+			parent[i] = i + 1;
```

Comment:
@Thomas-de-Rivaz it doesn't include any water, because of that it starts with id 1. But I agree that can be done in a more clear way.